### ISSUE TITLE
Export during publication finalization should not fail with NullPointerException and show dataset as "unpublished"

### ISSUE THREAD

**Issue by poikilotherm** [IQSS/dataverse#5759]
After searching I could not find a pre-existing issue report. Forgive me if this has been reported earlier.

## Prerequisites
1. Use S3 or similar file storage. Tempfile uploading needs to be successfull, but storage needs to fail. (E. g. non existing bucket)
2. Use v4.12. This is likely to exist in earlier versions, too.

## Steps to reproduce:
1. Create a new dataset and upload a file (it should not matter if adding the file during creation or afterwards)
2. Watch the upload fail and creation succeed (which is reported in the UI).
3. Try to publish the dataset. (File PID enabled by default).
4. See no error message, but log entries. Registration should complete (see my example [here](https://search.test.datacite.org/works/10.70012/fzj-data/2kpsdc)).
5. Dataset is still shown as "unpublished".

![grafik](https://user-images.githubusercontent.com/5468333/56034324-50126000-5d27-11e9-94c8-e4e9e91bbdb6.png)

See logs attached: [dataverse-5759-npe-export.txt](https://github.com/IQSS/dataverse/files/3073048/dataverse-5759-npe-export.txt). As the tempFile has been deleted, it is still `null` at [line 281](https://github.com/IQSS/dataverse/blob/b2180be1dd6a88cd89d17672b886e6a53722dab8/src/main/java/edu/harvard/iq/dataverse/export/ExportService.java#L281). This condition is not catched.

I can't tell if this leads to other unintended sideeffects later on.

**Comment by poikilotherm** [comment]
In addendum:
When fixing the S3 storage, publishing is succesfull. As a sideeffect, the identifier is changed, now being 
https://doi.org/10.70012/FZJ-DATA/OXRTWD (was https://doi.org/10.70012/FZJ-DATA/2KPSDC).

I tested this after uploading the missing file! It would need another test here... ;-)
The image had been uploaded under the old identifier 2KPSDC and I had to move it on the storage backend, otherwise it would give me an error 404 because of not finding it under the new identifier.
